Binary files wxorig/.DS_Store and wxnew/.DS_Store differ
Binary files wxorig/msw/.DS_Store and wxnew/msw/.DS_Store differ
diff -uNr wxorig/msw/registry.cpp wxnew/msw/registry.cpp
--- wxorig/msw/registry.cpp	2011-02-08 20:30:40.000000000 -0500
+++ wxnew/msw/registry.cpp	2011-03-18 12:39:46.000000000 -0400
@@ -370,6 +370,8 @@
   // it might be unexpected to some that this function doesn't open the key
   wxASSERT_MSG( IsOpened(), wxT("key should be opened in GetKeyInfo") );
 
+  DWORD SubKeys(0), MaxKeyLen(0), Values(0), MaxValueLen(0);
+
   m_dwLastError = ::RegQueryInfoKey
                   (
                     (HKEY) m_hKey,
@@ -377,19 +379,28 @@
                     NULL,           // (ptr to) size of class name buffer
                     RESERVED,
                     REG_PARAM
-                    pnSubKeys,      // [out] number of subkeys
+                    &SubKeys,      // [out] number of subkeys
                     REG_PARAM
-                    pnMaxKeyLen,    // [out] max length of a subkey name
+                    &MaxKeyLen,    // [out] max length of a subkey name
                     NULL,           // longest subkey class name
                     REG_PARAM
-                    pnValues,       // [out] number of values
+                    &Values,       // [out] number of values
                     REG_PARAM
-                    pnMaxValueLen,  // [out] max length of a value name
+                    &MaxValueLen,  // [out] max length of a value name
                     NULL,           // longest value data
                     NULL,           // security descriptor
                     NULL            // time of last modification
                   );
 
+  if (pnSubKeys)
+    *pnSubKeys = SubKeys;
+  if (pnMaxKeyLen)
+    *pnMaxKeyLen = MaxKeyLen;
+  if (pnValues)
+    *pnValues = Values;
+  if (pnMaxValueLen)
+    *pnMaxValueLen = MaxValueLen;
+
 #undef REG_PARAM
 
   if ( m_dwLastError != ERROR_SUCCESS ) {
